src/librustc/diagnostics.rs

```diff
@@ -112,6 +112,46 @@ reference when using guards or refactor the entire expression, perhaps by
 putting the condition inside the body of the arm.
 "##,
 
+E0152: r##"
+Lang items are already implemented in the standard library. Unless you are
+writing a free-standing application (e.g. a kernel), you do not need to provide
+them yourself.
+
+You can build a free-standing crate by adding `#![no_std]` to the crate
+attributes:
+
+#![feature(no_std)]
+#![no_std]
+
+See also https://doc.rust-lang.org/book/no-stdlib.html
+"##,
+
+E0158: r##"
+`const` and `static` mean different things. A `const` is a compile-time
+constant, an alias for a literal value. This property means you can match it
+directly within a pattern.
+
+The `static` keyword, on the other hand, only guarantees a fixed location in
+memory. This does not always mean that the value is constant. For example, a
+global mutex can be declared `static`, and it clearly does not stay the same.
+
+If you must match against a `static`, consider using a guard instead:
+
+static FORTY_TWO: i32 = 42;
+match Some(42) {
+    Some(x) if x == FORTY_TWO => ...
+    ...
+}
+"##,
+
+E0161: r##"
+In Rust, you can only move a value when its size is known at compile time.
+
+To work around this restriction, you can "hide" the value behind a pointer:
+either a reference (`&x`) or owned box (`box x`). Since a pointer has a fixed
```

Comment:
You can't call `Box::new()` with an unsized value.

For example, the following doesn't compile:

```rust
enum A {
    B(char),
    C([i32])
}

fn main() {
    Box::new(A::B('a'));
}
```

I chose not to add a note about feature gates, because:

1. It's quite a rare error -- most code shouldn't construct unsized values directly, and when they do, they tend to trigger E0277 ("trait Sized is not implemented") not this one

2. I don't want to add more churn when the syntax is stabilizing soon anyway